floyd/client/data.py

```diff
@@ -28,30 +25,30 @@ def __init__(self):
 
     def create(self, data):
         """
-        Create a temporary directory for the tar file that will be removed at the
-        end of the operation.
+        Create a temporary directory for the tar file that will be removed at
+        the end of the operation.
         """
         try:
             floyd_logger.info("Making create request to server...")
-            response = self.request("POST",
-                                    self.url,
-                                    data={"resumable": "true"})
-
-            floyd_logger.info("Done")
+            post_body = data.to_dict()
+            post_body["resumable"] = True
+            response = self.request("POST", self.url, json=post_body)
             return response.json().get("id")
         except FloydException as e:
             floyd_logger.info("Data create: ERROR! {}".format(e.message))
             return None
 
-    def new_tus_credentials(self, id):
+    def new_tus_credentials(self, data_id):
         try:
-            response = self.request("POST",
-                                    "{}new_credentials/".format(self.url),
-                                    data=json.dumps({"id": id}))
+            response = self.request(
+                "POST",
+                "%s%s/upload_credentials" % (self.url, data_id))
             data_dict = response.json()
             return (data_dict["data_upload_id"], data_dict["token"])
-        except FloydException as e:
-            floyd_logger.info("Error while fetching data upload credentials! {}".format(id, e.message))
+        except FloydException:
+            floyd_logger.exception(
```

Comment:
Yeah, I am changing this back. Ideally, we should have a `--debug` flag for the cli which turns on the exception logging so we can ask users to collect debug information for us.